scripts-available/CDB_CartodbfyTable.sql

```diff
@@ -893,9 +932,16 @@ BEGIN
   -- We can only avoid a rewrite if both the key and 
   -- geometry are usable
 
+  -- If cartodb_id column is of type text, cast it
+  IF idc.text_key_column THEN
+    sql := Format('ALTER TABLE %s ALTER cartodb_id TYPE int USING %I::integer', reloid::text, 'cartodb_id');
```

Comment:
this is precisely what we're trying to avoid... I'll explain to you the whole thing (todo el chiringuito)